dev/io.openliberty.checkpoint_fat_transaction/fat/src/io/openliberty/checkpoint/fat/TransactionScopedBeanTest.java

```diff
@@ -0,0 +1,183 @@
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.checkpoint.fat;
+
+import static org.junit.Assert.assertNotNull;
+
+import java.io.File;
+import java.security.AccessController;
+import java.security.PrivilegedExceptionAction;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.FileAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+
+import componenttest.annotation.ExpectedFFDC;
+import componenttest.annotation.Server;
+import componenttest.annotation.SkipIfCheckpointNotSupported;
+import componenttest.annotation.TestServlet;
+import componenttest.annotation.TestServlets;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.custom.junit.runner.Mode;
+import componenttest.custom.junit.runner.Mode.TestMode;
+import componenttest.rules.repeater.JakartaEE10Action;
+import componenttest.rules.repeater.JakartaEE9Action;
+import componenttest.rules.repeater.RepeatTests;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+import componenttest.topology.utils.HttpUtils;
+import io.openliberty.checkpoint.spi.CheckpointPhase;
+import transactionscopedtest.TransactionScopedBeanServlet;
+
+@RunWith(FATRunner.class)
+@SkipIfCheckpointNotSupported
+//@Mode(TestMode.FULL)
+public class TransactionScopedBeanTest extends FATServletClient {
+
+    static final String SERVER_NAME = "checkpointTransactionScopedBean";
+
+    @ClassRule
+    public static RepeatTests r = RepeatTests.withoutModification() //
+                    .andWith(new JakartaEE9Action().forServers(SERVER_NAME).fullFATOnly()) //
+                    .andWith(new JakartaEE10Action().forServers(SERVER_NAME).fullFATOnly());
+
+    static final String APP_NAME = "transactionscopedbean";
+    static final String SECOND_APP_NAME = "transactionscopedbeantwo";
+    static final String SERVLET_NAME = APP_NAME + "/transactionscopedbean";
+
+    @Server(SERVER_NAME)
+    @TestServlets({ @TestServlet(servlet = TransactionScopedBeanServlet.class, contextRoot = APP_NAME),
```

Comment:
Similar confusion for `@TestServlets`.  I'm not sure what is supposed to happen when we have `@Test` methods from the test class AND the test servlets.  Do both get run?

Reply:
I see the same behavior either way when the test method names in the Test class match those in the servlets.  Test methods run once.  I removed the `@TestServlet` from Test classes that do not rely on simplicity to run test methods defined in a servlet.